Simplenote/AccountDeletionHandler.swift

```diff
@@ -0,0 +1,17 @@
+import Foundation
+
+class AccountDeletionController {
```

Comment:
Mixed feelings about this controller!! (because it really does not add value). Maybe we should keep it, for consistency reasons (no other *Remote* is used directly in any ViewController).

If we do keep it:

1. The filename doesn't match the class name
2. We should move it to the `Controllers` logic folder?
3. The `SPTracker` invocation should go at the ViewController level.
3.1. Why: because `requestAccountDeletion` doesn't really have "knowledge" of whether this happened because of a Button Tap, or else.

Reply:
>Mixed feelings about this controller!! (because it really does not add value). Maybe we should keep it, for consistency reasons (no other Remote is used directly in any ViewController).

This is the thought process I went though.  Ultimately I decided to keep the controller so that the VC didn't know anything about the remote, but it also feels like a waste... I've made the requested changes.  

_Maybe_ we could look at combining the AccountDeletionController into AccountVerificationController for something like `AccountStateController`  I'm not really in love with the idea, but since I combined the account remotes it would at least be consistent.  WDYT?